File: functions/src/Task/entity/value_object/TaskType.ts
class taskType {
    private value: string;
  
    private constructor(value: string) {
      this.value = value;
    }
  
    static readonly WEEKLY = new taskType("weekly");
    static readonly DAILY = new taskType("daily");
    static readonly PROJECT = new taskType("project");
    static readonly WORK = new taskType("work");
    static readonly OTHER = new taskType("other");
  
    static fromString(value: string): taskType {
      switch (value.toLowerCase()) {
        case "weekly":
          return taskType.WEEKLY;
        case "daily":
          return taskType.DAILY;
        case "project":
          return taskType.PROJECT;
        case "work":
          return taskType.WORK;
        case "other":
          return taskType.OTHER;
        default:
          throw new Error(`Invalid taskType: ${value}`);
      }
    }
  
    public toString(): string {
      return this.value;
    }
  
    public equals(other: taskType): boolean {
      return this.value === other.value;
    }
  }
  
  export default taskType;
  

File: lib/model/value object/tasktype.dart
class TaskType {
  final String _value;

  const TaskType._(this._value);

  static const TaskType assignment = TaskType._("assignment");
  static const TaskType daily = TaskType._("daily");
  static const TaskType project = TaskType._("project");
  static const TaskType work = TaskType._("work");
  static const TaskType other = TaskType._("other");

  static const List<TaskType> values = [
    assignment,
    daily,
    project,
    work,
    other,
  ];

  static TaskType fromString(String value) {
    switch (value.toLowerCase()) {
      case 'assignment':
        return assignment;
      case 'daily':
        return daily;
      case 'project':
        return project;
      case 'work':
        return work;
      case 'other':
        return other;
      default:
        throw ArgumentError('Invalid TaskType: $value');
    }
  }

  @override
  String toString() => _value;

  @override
  bool operator ==(Object other) =>
      identical(this, other) ||
      other is TaskType && runtimeType == other.runtimeType && _value == other._value;

  @override
  int get hashCode => _value.hashCode;
}
